src/rez/package_order.py

```diff
@@ -147,26 +282,36 @@ def __init__(self, order_dict, default_order=None):
             default_order (PackageOrder): Orderer to apply to any packages
                 not specified in ``order_dict``.
         """
+        super().__init__(list(order_dict))
         self.order_dict = order_dict.copy()
         self.default_order = default_order
 
     def reorder(self, iterable, key=None):
-        try:
-            item = next(iter(iterable))
-        except:
+        package_name = self._get_package_name_from_iterable(iterable, key)
+        if package_name is None:
             return None
 
-        key = key or (lambda x: x)
-        package = key(item)
-
-        orderer = self.order_dict.get(package.name)
+        orderer = self.order_dict.get(package_name)
         if orderer is None:
             orderer = self.default_order
         if orderer is None:
             return None
 
         return orderer.reorder(iterable, key)
 
+    def sort_key_implementation(self, package_name, version):
+        orderer = self.order_dict.get(package_name)
+        if orderer is None:
+            if self.default_order is not None:
+                orderer = self.default_order
+            else:
+                # shouldn't get here, because applies_to should protect us...
+                raise RuntimeError(
+                    "package family orderer %r does not apply to package family %r",
+                    (self, package_name))
```

Comment:
```suggestion
            
            if self.default_order is None:
                # shouldn't get here, because applies_to should protect us...
                raise RuntimeError(
                    "package family orderer %r does not apply to package family %r",
                    (self, package_name))

            orderer = self.default_order
```
We could also potentially use an `assert`.

Reply:
Yeah, assert would be fine.  What you've suggested is also fine though.

Reply:
Sorry, I thought your change had been added to the batch.  Merged in now.